processing/src/main/java/org/apache/druid/frame/key/ClusterBy.java

```diff
@@ -58,6 +59,21 @@ public ClusterBy(
     if (bucketByCount < 0 || bucketByCount > columns.size()) {
       throw new IAE("Invalid bucketByCount [%d]", bucketByCount);
     }
+
+    // Key must be 100% sortable or 100% nonsortable. If empty, call it sortable.
+    boolean sortable = true;
+
+    for (int i = 0; i < columns.size(); i++) {
+      final KeyColumn column = columns.get(i);
+
+      if (i == 0) {
+        sortable = column.order().sortable();
+      } else if (sortable != column.order().sortable()) {
+        throw new IAE("Cannot mix sortable and unsortable key columns");
+      }
+    }
+
+    this.sortable = sortable;
```

Comment:
Code nit, could also do:

```
if (columns.size == 0) {
  // no columns means that it's sortable/sorted as they are effectively all sorted by the same non-existant, constant value.
  this.sortable = true;
} else {
  boolean sortable = columns.get(0).order().sortable();
  for (int i = 1; i < columns.size; ++i)  {
    if (sortable != column.order().sortable()) {
      throw new IAE("Cannot mix sortable and unsortable key columns");
    }
  }
}
```

Reply:
I kept it as-is since they both seem equally good to me and I'm lazy 🙂